mongo/integration/mtest/mongotest.go

```diff
@@ -696,9 +698,14 @@ func verifyServerParametersConstraints(serverParameters map[string]bson.RawValue
 	return nil
 }
 
-func verifyAuthConstraint(expected *bool) error {
-	if expected != nil && *expected != testContext.authEnabled {
-		return fmt.Errorf("test requires auth value: %v, cluster auth value: %v", *expected, testContext.authEnabled)
+func verifyAuthConstraint(expectedAuth *bool, expectedAuthEnabled *bool) error {
+	// Most tests in the unified format have runOn.auth to indicate whether the
```

Comment:
The Unified Test Format only defines `auth` in the [runOnRequirement block](https://github.com/mongodb/specifications/blob/df6be82f865e9b72444488fd62ae1eb5fca18569/source/unified-test-format/unified-test-format.rst#runonrequirement).

The SDAM integration tests appear to be the only tests that [specify `authEnabled`](https://github.com/mongodb/specifications/blob/df6be82f865e9b72444488fd62ae1eb5fca18569/source/server-discovery-and-monitoring/tests/README.rst#test-format).

Can you make this comment more precise? It may be helpful to readers to know exactly which tests have the difference.